add postgresql_script resource (#18)
doctolib/terraform-provider-postgresql: website/docs/r/postgresql_script.html.markdown

> ---

Reviewer: I think there are other places to edit to reference this page.

layout: "postgresql"
page_title: "PostgreSQL: postgresql_script"
sidebar_current: "docs-postgresql-resource-postgresql_script"
description: |-

>   Execute a SQL scipt

Reviewer: ```suggestion
  Execute a SQL script
```

---

# postgresql\_script

> The ``postgresql_script`` execute a script given as parameter.

Reviewer: ```suggestion
The ``postgresql_script`` execute a script given as parameter. This script will be executed each time it changes.
```

> The ``postgresql_script`` execute a script given as parameter. This script will be executed each time it changes.

Reviewer: ```suggestion
The ``postgresql_script`` execute a script given as parameter. This script will be executed each time it changes.

If one command of the batch fails, the provider will send a `ROLLBACK` command to the database, and retry, according to the tries / backoff_delay configuration.
```

## Usage

```hcl
resource "postgresql_script" "foo" {
  commands = [
    "command 1",
    "command 2"
  ]
  tries = 1

>   timeout = 1

Reviewer: ```suggestion
  backoff_delay = 1
```

}
```

## Argument Reference

* `commands` - (Required) An array of commands to execute, one by one.
* `tries` - (Optional) Number of tries of a command before raising an error.

> * `timeout` - (Optional) Time in second between two failing commands.

Reviewer: ```suggestion
* `backoff_delay` - (Optional) In case of failure, time in second to wait before a retry.
```

## Examples

Revoke default accesses for public schema:

```hcl
resource "postgresql_script" "foo" {
  commands = [
    "BEBIN",
    "SELECT * FROM table",
    "COMMIT"
  ]
  tries = 3

>   timeout = 1

Reviewer: ```suggestion
  backoff_delay = 1
```
